src/ck-ldb/TreeLevel.h

```diff
@@ -552,7 +552,10 @@ class RootLevel : public LevelLogic
       decisions.resize(num_children);
       decisions[0] = migMsg;
       for (int i = 1; i < num_children; i++)
-        decisions[i] = (TreeLBMessage*)CkCopyMsg((void**)&migMsg);
+      {
+        CmiReference(UsrToEnv(migMsg));
```

Comment:
We should really add a CkMsgReference() API (and perhaps a CkMsgSetUsed) like this:

void *CkMsgReference(void *msg) {
    CmiReference(UsrToEnv(msg));
    return msg;
}

Then this can become  'decisions[i] = CkMsgReference(migMsg);'

Can be done in a separate patch

Reply:
Yeah, good idea, I've made #2862.